### ISSUE TITLE
Dynamic link pasted in to URL bar redirects to home page, menu link works as expected

### ISSUE THREAD

**Issue by kerrsmith** [nuxt/nuxt.js#5967]
### Version

[v2.8.1](https://github.com/nuxt.js/releases/tag/v2.8.1)

### Reproduction link

 I have now deleted the site from my test server

### Steps to reproduce

use npm run generate
upload files to server
paste a link to a dynamic page in to the URL bar - you are taken to homepage
use link on page and the page opens as expected

### What is expected ?

The generated page should open the correct page when using generated code the same as when using npm run dev

### What is actually happening?

When using generated code directly pasted links redirect to the homepage, using on site links and the page loads as expected.

If you refresh a generated page it also goes to homepage.

Using npm run dev and the pages work as expected.

### Additional comments?

When I run my code using npm run dev and I paste a URL to a dynamic page the page loads as expected, when I generate the code (npm run generate), upload to my server and then try to go to the URL directly I get redirected to the home page. Access to the dynamic pages via the menus on the site work as expected.

If I hit reload on a dynamic page it also loads the homepage.

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c9396">#c9396</a>)</em></sub></div>

**Comment by pimlie** [comment]
The redirect might be caused by a webserver configuration, as eg
 https://www.wales.cloudns.org/deckchairs/traditional_multi_stripe/index.html does work so the file exists. Whats strange is that the url with index.html doesnt load any items, small chance but could be related

**Comment by kerrsmith** [comment]
It is very strange as it was working and I have not changed the server configuration. I did add an .htaccess file to redirect all the site's old pages that no longer exist to the homepage but I removed this and the redirection when reloading a page still happens.

I am going to go back through a copy of my code and try to use git bisect to see what changed and caused it to stop working as it used to.

Thanks for the suggestion - I will certainly investigate this as well as possible cause though.

**Comment by kerrsmith** [comment]
I have now solved the issue thanks to https://github.com/nuxt/nuxt.js/issues/5967#issuecomment-504647266

It was indeed a server configuration due to the .htaccess file I added a while ago. When I tested the code by removing this file the refresh still caused the page to go back to the homepage but when I did a 'shift' refresh then it started working so there must have been a caching issue.

My plan with the .htaccess file was to redirect all links that pointed to old now non-existent pages to the homepage, the contents of this file were:

RewriteEngine on
RewriteCond %{REQUEST_FILENAME} !-f
RewriteRule . / [L,R=301]

As I can now see I had not included a check for directories:

RewriteCond %{REQUEST_FILENAME} !-d

I have added this extra line to check for directories and the site now works as it used to.

Thanks again for the suggestion - I really appreciate it.
